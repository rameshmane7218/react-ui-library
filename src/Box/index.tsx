export { Box, BoxProps } from './Box';
