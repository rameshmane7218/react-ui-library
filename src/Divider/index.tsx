export { Divider, HrProps } from './Divider';
