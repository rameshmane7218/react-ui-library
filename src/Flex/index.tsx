export { Flex, FlexProps } from './Flex';
