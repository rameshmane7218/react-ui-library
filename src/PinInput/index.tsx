export { InputField, InputFieldProps } from './InputField';
export { PinInput, PinInputProps } from './PinInput';
export { PinInputExample } from './PinInputExample';
