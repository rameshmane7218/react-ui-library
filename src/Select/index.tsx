export { Select, SelectProps } from './Select';
export { Option, OptionProps } from './Option';
