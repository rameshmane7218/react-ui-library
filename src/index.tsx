export * from './Button';
export * from './Thing';
