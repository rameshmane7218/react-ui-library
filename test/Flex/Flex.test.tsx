import React from 'react';
import { render } from '@testing-library/react';
import { Box, Flex } from '../../src';

describe('Flex Component', () => {
  it('should render Flex component without crashing', () => {
    render(
      <Flex>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Nihil
        reiciendis mollitia quas deserunt enim accusamus! Minima dolorum vero
        possimus magni mollitia quam repudiandae totam, amet nisi nobis eum
        consectetur nam!
      </Flex>
    );
  });
  it('render all flex direction', () => {
    const { asFragment } = render(
      <Flex flexDirection="column">
        <Flex flexDirection="row" gap={'30px'}>
          <Box>box1</Box>
          <Box>box2</Box>
        </Flex>
        <Flex flexDirection="row-reverse">
          <Box>box1</Box>
          <Box>box2</Box>
        </Flex>
        <Flex flexDirection="column">
          <Box>box1</Box>
          <Box>box2</Box>
        </Flex>
        <Flex flexDirection="column-reverse">
          <Box>box1</Box>
          <Box>box2</Box>
        </Flex>
      </Flex>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('render justifyContent properly', () => {
    const { asFragment } = render(
      <Flex flexDirection="column">
        <Flex justifyContent="space-between">
          <Box>box1</Box>
          <Box>box2</Box>
        </Flex>
        <Flex justifyContent="space-around">
          <Box>box1</Box>
          <Box>box2</Box>
        </Flex>
        <Flex justifyContent="space-evenly">
          <Box>box1</Box>
          <Box>box2</Box>
        </Flex>
        <Flex justifyContent="start">
          <Box>box1</Box>
          <Box>box2</Box>
        </Flex>
        <Flex justifyContent="end">
          <Box>box1</Box>
          <Box>box2</Box>
        </Flex>
      </Flex>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
