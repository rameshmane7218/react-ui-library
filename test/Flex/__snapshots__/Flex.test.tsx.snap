// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex Component render all flex direction 1`] = `
<DocumentFragment>
  <div
    class="sc-dkrFOg gDwFTQ"
    width="100%"
  >
    <div
      class="sc-dkrFOg jgclOZ"
      width="100%"
    >
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box1
      </div>
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box2
      </div>
    </div>
    <div
      class="sc-dkrFOg gOZmkh"
      width="100%"
    >
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box1
      </div>
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box2
      </div>
    </div>
    <div
      class="sc-dkrFOg gDwFTQ"
      width="100%"
    >
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box1
      </div>
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box2
      </div>
    </div>
    <div
      class="sc-dkrFOg ejOEiZ"
      width="100%"
    >
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box1
      </div>
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box2
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex Component render justifyContent properly 1`] = `
<DocumentFragment>
  <div
    class="sc-dkrFOg gDwFTQ"
    width="100%"
  >
    <div
      class="sc-dkrFOg fiUeUj"
      width="100%"
    >
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box1
      </div>
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box2
      </div>
    </div>
    <div
      class="sc-dkrFOg dWWPIW"
      width="100%"
    >
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box1
      </div>
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box2
      </div>
    </div>
    <div
      class="sc-dkrFOg eEjfs"
      width="100%"
    >
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box1
      </div>
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box2
      </div>
    </div>
    <div
      class="sc-dkrFOg hSqOaI"
      width="100%"
    >
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box1
      </div>
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box2
      </div>
    </div>
    <div
      class="sc-dkrFOg cigUhb"
      width="100%"
    >
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box1
      </div>
      <div
        class="sc-gswNZR fnnikn"
        width="none"
      >
        box2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
